internal/provider/common.go

```diff
@@ -41,26 +41,25 @@ func getExponentialBackoff(initial_interval time.Duration, multiplier float64, m
 	return b
 }
 
-func waitDeployment(ctx context.Context, nodeClient *client.NodeClient, deploymentID uint64, version uint32) error {
-	last_progress := Progress{time.Now(), 0}
+func waitDeployment(ctx context.Context, nodeClient *client.NodeClient, deploymentID uint64, version uint32, workloadVersions map[string]uint32) error {
+	lastProgress := Progress{time.Now(), 0}
 
-	deployment_error := backoff.Retry(func() error {
-		var version_err error = errors.New("deployment version not updated on node")
+	deploymentError := backoff.Retry(func() error {
+		var deploymentVersionError error = errors.New("deployment version not updated on node")
 		done := true
-		state_Ok := 0
+		stateOk := 0
 
 		sub, cancel := context.WithTimeout(ctx, 10*time.Second)
 		defer cancel()
 		dl, err := nodeClient.DeploymentGet(sub, deploymentID)
 		if err != nil {
 			return backoff.Permanent(err)
 		}
-
 		if dl.Version == version {
-			version_err = nil
+			deploymentVersionError = nil
 			for idx, wl := range dl.Workloads {
-				if wl.Result.State == gridtypes.StateOk {
-					state_Ok++
+				if _, ok := workloadVersions[wl.Name.String()]; ok && wl.Version == workloadVersions[wl.Name.String()] && wl.Result.State == gridtypes.StateOk {
```

Comment:
In this scenario;
- a deployment is created, and all is ok.
- for some reason, a workload failed and its state became `error` on the node
- a deployment update is initiated using terraform, and this workload is changed
- at an instance, the deployment version is latest while the workload version is still the old one
- the second branch is entered because the state is error, but this shouldn't happen since this is still the old workload